chapters/location_component_interface.adoc

```diff
@@ -0,0 +1,98 @@
+// Copyright 205 The Khronos Group, Inc.
+// SPDX-License-Identifier: CC-BY-4.0
+
+ifndef::chapters[:chapters:]
+ifndef::images[:images: images/]
+
+[[location-and-component-interface]]
+= Location and Component Interface
+
+This chapter is an overview of the link:https://docs.vulkan.org/spec/latest/chapters/interfaces.html#interfaces-iointerfaces-locations[Location and Component Assignment] chapter to help give examples, especially around some of the more extreme edge cases.
+
+The simplest way to think about a `Location` is that it is made up of four 32-bit `Components`. This means a `vec4`/`float4`/`uvec4`/etc will fit perfectly in a single `Location`.
+
+`Location` are used for both the `Input` and `Output` to interface between shaders stages when possible.
+
+== Basic Example
+
+Here is a basic example:
+
+[source,glsl]
+----
+layout(location=0) in vec4 a;
+layout(location=1, component = 0) in vec2 b;
+layout(location=1, component = 2) in float c;
```

Comment:
Somewhere this could do with an explicit call-out that the vec2 consumes two components.